src/main/java/data_access/UserDataAccessObject.java

```diff
@@ -18,8 +17,8 @@
 public class UserDataAccessObject implements ClubSignupUserDataAccessInterface, StudentSignupUserDataAccessInterface,
         ClubLoginDataAccessInterface, StudentLoginDataAccessInterface, ClubCreatePostUserDataAccessInterface {
 
-    private final ArrayList<Student> studentArrayList = new ArrayList<>();
-    private final ArrayList<Club> clubArrayList = new ArrayList<>();
+    private final DataStoreArrays<Student> studentArrayList = new DataStoreArrays<>();
+    private final DataStoreArrays<Club> clubArrayList = new DataStoreArrays<>();
```

Comment:
Concerns with SOLID principles. See my main review text for a larger explanation.